fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStoreIndexTest.java

```diff
@@ -2093,29 +2094,64 @@ public void boundaryPrimaryKeys() throws Exception {
-                        indexer.buildEndpoints().thenCompose(range ->
-                                recordStore.getPrimaryKeyBoundaries(range.getLow(), range.getHigh()).asList()));
-
-                logger.info("The boundary primary keys are " + boundaryPrimaryKeys);
-
-                assertTrue(boundaryPrimaryKeys.size() > 2,
-                        "the test is meaningless if the records are not across boundaries");
-                assertThat( boundaryPrimaryKeys.get(0), greaterThanOrEqualTo(Tuple.from(-25L * 39)));
-                assertThat( boundaryPrimaryKeys.get(boundaryPrimaryKeys.size() - 1), lessThanOrEqualTo(Tuple.from(24L * 39)));
-                assertEquals(boundaryPrimaryKeys.stream().sorted().distinct().collect(Collectors.toList()), boundaryPrimaryKeys,
-                        "the list should be sorted without duplication.");
-                for (Tuple boundaryPrimaryKey : boundaryPrimaryKeys) {
-                    assertEquals(1, boundaryPrimaryKey.size(), "primary keys should be a single value");
-                }
-            }
+                    .build();
+
+            boundaryPrimaryKeys = recordStore.context.asyncToSync(FDBStoreTimer.Waits.WAIT_GET_BOUNDARY,
+                    indexer.buildEndpoints().thenCompose(range ->
+                            recordStore.getPrimaryKeyBoundaries(range.getLow(), range.getHigh()).asList()));
+
+            logger.info("The boundary primary keys are " + boundaryPrimaryKeys);
+
+            commit(context);
+        }
+
+        int boundaryPrimaryKeysSize = boundaryPrimaryKeys.size();
+        assertTrue(boundaryPrimaryKeysSize > 2,
+                "the test is meaningless if the records are not across boundaries");
+        assertThat( boundaryPrimaryKeys.get(0), greaterThanOrEqualTo(Tuple.from(-25L * 39)));
+        assertThat( boundaryPrimaryKeys.get(boundaryPrimaryKeysSize - 1), lessThanOrEqualTo(Tuple.from(24L * 39)));
+        assertEquals(boundaryPrimaryKeys.stream().sorted().distinct().collect(Collectors.toList()), boundaryPrimaryKeys,
+                "the list should be sorted without duplication.");
+        for (Tuple boundaryPrimaryKey : boundaryPrimaryKeys) {
+            assertEquals(1, boundaryPrimaryKey.size(), "primary keys should be a single value");
+        }
+
+
+        // Test splitIndexBuildRange.
+        assertEquals(1, indexer.splitIndexBuildRange(Integer.MAX_VALUE, Integer.MAX_VALUE).size(),
+                "the range is not split when it cannot be split to at least minSplit ranges");
+        checkSplitIndexBuildRange(boundaryPrimaryKeysSize / 2, boundaryPrimaryKeysSize, indexer);
+        checkSplitIndexBuildRange(boundaryPrimaryKeysSize / 2, boundaryPrimaryKeysSize + 1, indexer); // Exactly one range for each split
+        checkSplitIndexBuildRange(boundaryPrimaryKeysSize / 2, boundaryPrimaryKeysSize + 2, indexer);
```

Comment:
I think this could use the following additional test cases:

```java
 checkSplitIndexBuildRange(boundaryPrimaryKeysSize / 2, Integer.MAX_VALUE, indexer); // to test that integer overflow isn't a problem
 checkSplitIndexBuildRange(boundaryPrimaryKeysSize / 2, boundaryPrimaryKeysSize - 2, indexer); // to test splitting into fewer than the default number of split points
 checkSplitIndexBuildRange(1, 2, indexer); // to test splitting into fewer than the default number of split points
```

It's also possible that `checkSplitIndexBuildRange` should have a, like, "expected split ranges" parameter if it's knowable.

Reply:
Done. And it catches a glitch in calculating stepSize:)

Reply:
Well, good to know my pedantry caught something for once...